### ISSUE TITLE
Attacking a Wolf Causes the Server to Crash

### ISSUE THREAD

**Issue by Frownigami1** [cuberite/cuberite#3437]
When running Minecraft 1.10, attacking a wolf (tamed or otherwise) will crash the server. 

Happens when:
     - Melee attacking a wolf.
     - Shooting a wolf with a bow.

Does not happen when:
     - Mobs damage a wolf.
     - Wolves take fall damage.
     - Wolves take damage from environmental hazards like cacti, TNT, lava, or fire (even if the player places   
       them).
     - Players ignite a wolf with flint and steel.

So in short the server only crashes when the player directly does damage to a wolf.

**Comment by madmaxoft** [comment]
I confirm this behavior, there's a problem in the wolf code assuming the current target is always assigned.
